python/lsst/daf/butler/remote_butler/_remote_butler.py

```diff
@@ -629,18 +598,21 @@ def _get_url(self, path: str, version: str = "v1") -> str:
         path : `str`
             The full path to the endpoint.
         """
-        return f"{version}/{path}"
+        slash = "" if self._server_url.endswith("/") else "/"
```

Comment:
Can we force `self._server_url` to always have the `/` so we don't need to check every time on get?

Reply:
Yeah I had considered that but I was concerned someone might assign the server URL from somewhere else, forgetting the slash, and it would break intermittently.  By putting this check here we enforce it at the point where the code depends on it having a slash.

Reply:
Isn't `_server_url` private?

Reply:
Yes but RemoteButler is already almost 700 lines long and we haven't implemented 95% of its methods yet.  I don't see any reason not to enforce correctness at the point of use instead of adding a comment like "make extra sure this variable always has a slash at the end of it!"

Reply:
I think I'm missing something. `_server_url` is set in one place in the constructor so my suggestion was to move this fixup line up to that point and normalize the string there. Then it's fixed once and never needs to be tested again.

I wasn't suggesting we document that all places that construct a RemoteButler must ensure the / is added.

Reply:
Currently it is, but with the rate that these constructors are proliferating we may end up with it getting set more than one place.  

The only place that must have a slash is _get_url, so what is the benefit to doing this logic in the constructor?  The constructor is plenty busy with other issues already.